File: aspnet-in-practice/08-09/VB/CoolMVCBlog/Utils/EmailAddressAttribute.vb
Imports System.ComponentModel.DataAnnotations

Public Class EmailAddressAttribute
    Inherits RegularExpressionAttribute

    Friend Const EmailRegex As String =
        "\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,4}\b"

    Public Sub New()
        MyBase.New(EmailRegex)
    End Sub
End Class

File: aspnet-in-practice/08-09/CS/CoolMVBlog/Utils/EmailAddressAttribute.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.ComponentModel.DataAnnotations;

namespace CoolMvcBlog.Utils
{
    public class EmailAddressAttribute : RegularExpressionAttribute
    {
        internal const string EmailRegex = @"\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,4}\b";

        public EmailAddressAttribute(): base(EmailRegex)
        {
            this.ErrorMessage = "Please provide a valid email address";
        }
    }
}